### PR TITLE
Async API Lesson - Added Learning Outcomes, Knowledge Checks, Links

### PR DESCRIPTION
**Pull request by digidub**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The PR title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 
#### 1.Describe the changes made and include why they are necessary or important:

Added Learning Outcomes, Knowledge Checks, and Links based on the lesson's outline. Please let me know if you'd like any changes to be made.

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

#22472

### CODE REVIEW

File: javascript/async-apis/APIs.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 One of the most powerful things a web developer can do is fetching data from a server and displaying it creatively on their site. In many cases, the server solely exists for that specific site. The server could contain blog posts, user data, high scores for a game or anything else. In other cases, the server is an open service that serves data to anyone that wants to use it (i.e. weather data or stock prices). In either case, the methods of accessing and then using that data are essentially the same.
 
+### Learning Outcomes
+By the end of this lesson, you should be able to:
+
+ - Explain what an API is
+ - Explain how access to an API broadly works
```
Comment (kashura) [inline javascript/async-apis/APIs.md:9]: Broadly seems in the wrong place. Did you mean to put it after Explain?

File: javascript/async-apis/APIs.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
...
+ - Explain how you fetch and extract data from an API
```
Comment (kashura) [inline javascript/async-apis/APIs.md:10]: Please replace "you" with "to".

File: javascript/async-apis/APIs.md [hunk 3]
```diff
@@ -2,6 +2,14 @@
...
+ - Explain why your API request might be blocked by your browser, and how to fix this
```
Comment (kashura) [inline javascript/async-apis/APIs.md:11]: Please replace "your" with "the" in this part "blocked by your browser".

**Review by kashura — CHANGES_REQUESTED** [review]

**Comment by kashura** [comment]
Just a few minor comments. Thank you!

File: javascript/async-apis/APIs.md (hunk 1 above)
Comment (digidub) [reply javascript/async-apis/APIs.md:9]: I actually started with it where you've suggested it and then changed it 😁 I'll revert.

**Comment by digidub** [comment]
Thanks, requested changes have been made.

**Review by kashura — APPROVED** [review]
Thank you for addressing requested changes :-)

**Pull request merged**